napari/utils/settings/_tests/test_settings.py

```diff
@@ -113,3 +114,24 @@ def test_settings_model(settings):
     with pytest.raises(pydantic.error_wrappers.ValidationError):
         # Should be a valid string
         settings.application.theme = "vaporwave"
+
+
+def test_custom_theme_settings(settings):
+    # See: https://github.com/napari/napari/issues/2340
+    custom_theme_name = "blue"
+
+    # No theme registered yet, this should fail
+    with pytest.raises(pydantic.error_wrappers.ValidationError):
+        settings.application.theme = custom_theme_name
```

Comment:
If this solves issue #2340 then why we expect failiture? @sofroniewn @goanpeca

Reply:
You have to register the theme first with `register_theme`, but it should be 

```python
    register_theme(custom_theme_name, blue_theme)
```

not 

```python
    register_theme(custom_theme_name, custom_theme_name)
```

below (line 134). Maybe @goanpeca can fix that in a follow up PR

Reply:
I see now why it expects failure. But I see something strange. When I run test only on this file everything pass. But when I run all tests I got.

```
FAILED [ 97%]
napari/utils/settings/_tests/test_settings.py:118 (test_custom_theme_settings)
settings = <napari.utils.settings._manager.SettingsManager object at 0x7f81ae0d3e50>

    def test_custom_theme_settings(settings):
        # See: https://github.com/napari/napari/issues/2340
        custom_theme_name = "blue"
    
        # No theme registered yet, this should fail
        with pytest.raises(pydantic.error_wrappers.ValidationError):
>           settings.application.theme = custom_theme_name
E           Failed: DID NOT RAISE <class 'pydantic.error_wrappers.ValidationError'>

napari/utils/settings/_tests/test_settings.py:125: Failed****
```

maybe it is connected with the execution of examples?

Reply:
No. I meet this on the last master (during the local test of #2422)

Reply:
It still happens. I think that it is connected with global `_theme` object (inside `napari.utils.theme`) 

Both pointed test and `new_theme.py` example register theme `blue`. The simplest workaround is to change the name in one of them. 

For me the better solution is to restore default values of global variables on the teardown of the test instance.